src/tools/miri/src/shims/foreign_items.rs

```diff
@@ -900,7 +900,7 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
             // Fallback to shims in submodules.
             _ => {
                 // Math shims
-                #[expect(irrefutable_let_patterns)]
+                #[allow(irrefutable_let_patterns)]
```

Comment:
The entire point of using `expect` here is to not have spurious attributes around when this lint goes away. Changing this to `allow` defeats that purpose and leaves the code to bitrot. Please don't do that.

`cfg_attr(bootstrap, expect(...))` should work.

Reply:
For cases where your PR makes the lint go away, changing the attribute to `cfg_attr(bootstrap, expect(irrefutable_let_patterns))` seems like a good idea.

But please don't refactor any other code as part of this PR, just keep the `allow` if the  lint still fires there with your PR.